### PR TITLE
Feature/countries tag filters

### PR DESCRIPTION
**Pull request by edbrett**
screen shot 2018-08-13 at 16 58 40

### Overview
Adds the countries tags selector feature to the datasets menu component! The feature requires that the user can tag countries on the map menu in order to filter the datasets and searches. This PR adds the initial functionality to the menu, refactoring the `CountryDataProvider` to remove the geostore dependancy and creates a new `GeostoreProvider` component for managing this more atomically. 

Additionally the menu component is refactored to handle the data structure change, binding the `MapMenu` component to the url for persistence. See the `menu-selectors.js` to see the parity between the actions and selectors to the `map-selectors.js`.

### Testing
You can add and remove countries with tags, otherwise functionality is the same. Map bounding box based on interaction and layers filtering will come later.

### tldr
- Add `<Pill />` component for country tag menu
- Refactor `CountryDataProvider` to new `GeostoreProvider`
- Update `MapMenu` to use updated store structure
- Lots of styles fixes for this excellent set of changes

### CODE REVIEW

File: app/javascript/components/ui/icon/icon-component.jsx [hunk 1]
```diff
@@ -3,15 +3,20 @@ import PropTypes from 'prop-types';
 
 import './icon-styles.scss';
 
-const Icon = ({ icon, className }) => (
-  <svg className={`c-icon ${className}`} viewBox={icon.viewBox || '0 0 32 32'}>
+const Icon = ({ icon, className, onClick }) => (
+  <svg
+    className={`c-icon ${className}`}
+    viewBox={icon.viewBox || '0 0 32 32'}
+    onClick={onClick}
```
Comment (sorodrigo) [inline app/javascript/components/ui/icon/icon-component.jsx]: but why

File: app/javascript/components/ui/pill/component.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import closeIcon from 'assets/icons/close.svg';
+import './styles.scss';
+
+const Button = props => {
+  const { label, onRemove } = props;
+
+  return (
+    <div className="c-pill">
+      {label}
+      <Icon icon={closeIcon} onClick={onRemove} />
```
Comment (sorodrigo) [inline app/javascript/components/ui/pill/component.jsx]: should be in a button instead of a svg

File: app/javascript/components/ui/icon/icon-component.jsx (hunk 1 above)
Comment (edbrett) [reply app/javascript/components/ui/icon/icon-component.jsx]: So I can click the Icon.

File: app/javascript/pages/map/menu/components/sections/datasets/component.jsx [hunk 3]
```diff
@@ -1,20 +1,75 @@
...
       <div className="c-datasets">
+        <div className="countries-selection">
+          <span className="sub-title">country-specific data</span>
+          <div className="pills">
+            {selectedCountries &&
+              selectedCountries.map(c => (
+                <Pill
+                  key={c.value}
+                  label={c.label}
+                  onRemove={() => {
```
Comment (sorodrigo) [inline app/javascript/pages/map/menu/components/sections/datasets/component.jsx]: use a class method instead

Comment (edbrett) [reply app/javascript/pages/map/menu/components/sections/datasets/component.jsx]: Ah balls. Indeed. I forgot to move these out. Thanks.

File: app/javascript/pages/map/menu/components/sections/datasets/component.jsx [hunk 4]
```diff
@@ -1,20 +1,75 @@
...
+                    const newCountries = remove(
+                      selectedCountries,
+                      sc => sc.value !== c.value
```
Comment (sorodrigo) [inline app/javascript/pages/map/menu/components/sections/datasets/component.jsx]: why not use `selectedCountries.filter` instead?

Comment (edbrett) [reply app/javascript/pages/map/menu/components/sections/datasets/component.jsx]: Because I am a fool.

File: app/javascript/providers/geostore-provider/actions.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import { getGeostoreProvider } from 'services/geostore';
+
+export const setGeostoreLoading = createAction('setGeostoreLoading');
+export const setGeostore = createAction('setGeostore');
+
+export const getGeostore = createThunkAction(
+  'getGeostore',
+  (country, region, subRegion) => (dispatch, state) => {
+    if (!state().geostore.loading) {
```
Comment (sorodrigo) [inline app/javascript/providers/geostore-provider/actions.js]: maybe add a check for HMR?

File: app/javascript/components/ui/icon/icon-component.jsx (hunk 1 above)
Comment (sorodrigo) [reply app/javascript/components/ui/icon/icon-component.jsx]: I'd wrap the icon in a `<button>` and add the listener there. Adding it to the svg is weird, I'm sure it not super accesible

File: app/javascript/pages/map/menu/menu-component.jsx [hunk 6]
```diff
@@ -46,7 +48,9 @@ class Menu extends PureComponent {
                     <button
                       className="c-map-menu__item-button"
                       onClick={() => {
-                        setSelectedSection(slug);
+                        setMenuSettings({
```
[the remark this answers is not in the document]
Comment (edbrett) [reply app/javascript/pages/map/menu/menu-component.jsx]: I dont understand this one?

[the remark this answers is not in the document]
Comment (edbrett) [reply app/javascript/pages/map/menu/menu-component.jsx]: ah sorry I get it now. I read it weird.
